providers/azure.go

```diff
@@ -277,6 +277,10 @@ func (p *AzureProvider) extractClaimsIntoSession(ctx context.Context, session *s
 		session.Groups = s.Groups
 	}
 
+	if s.Roles != nil {
+		session.Roles = s.Roles
+	}
```

Comment:
So instead of extending the Session, which should stay as generic as possible, maybe we should introduce a custom `Authorize()` implementation for the azure provider. In this new `Authorize` method you could extract the claims and check for the roles you provide as part of the `AzureOptions` struct.

https://github.com/oauth2-proxy/oauth2-proxy/blob/fc6e7fdbd11f979de2c4e5c1d3d7eb8b214566be/providers/providers.go#L21-L32


@JoelSpeed what's your opinion on this? Should we keep the generic approach proposed in this PR or change it towards a more Azure / Microsoft Entra focused implementation?

Reply:
@tuunit & @JoelSpeed what's your thoughts?
We'd prefer not to keep a clone running for too long that we also need to maintain.